src/conn.c

```diff
@@ -1137,17 +1137,29 @@ _resendSubscriptions(natsConnection *nc)
 
         adjustedMax = 0;
         natsSub_Lock(sub);
+        if (sub->libDlvWorker != NULL)
```

Comment:
There are macro for lock/unlock of this in `sub.c`. Maybe we move the definition of these in `sub.h` (we include this header in `conn.c`), and so you could replace these 2 lines with:
```
SUB_DLV_WORKER_LOCK(sub);
```

Reply:
@kozlovic I stayed away from further exposing these macros. I am about to submit a re-factor (just passed all tests now without kicking!) that hopefully eliminates the need for double-locking, leaving the worker lock almost exclusively to protect the message queue. Ditto for the `sub->closed` comment.

It's ok with me if we keep this PR "hanging" for now until you see the upcoming solution, and then we can decide which way to fix? I just want to have it as my base branch to get the CI/unrelated test fixes out of the way.

Reply:
Not sure I understand the last part, that is, do you want to merge this PR to be the base for the other work or the other work will be the (re)base for the CI changes. Either way, I am going to LGTM this PR so that you proceed as you wish.